### DISCUSSION TITLE
Does Oembed 1.3.19 Support PHP 8?

### DISCUSSION THREAD

**Discussion by mmoss128** [wrav/oembed#159] [General]
I am working on upgrading my Craft environment from 3 to 4, and I have been playing with setting up PHP 8.0/8.3 prior to the upgrade. I noticed that the fields are no longer rendering if Caching is enabled. I have tried some debugging to see if this is something on myside. Just want to see if there is an issue with PHP 8.x+ which would cause this or if I should continue debugging.

**Comment by reganlawton** [comment]
@mmoss128 Have you tried upgrading the plugin to 2.x? If you’re upgrading CraftCMS I don't believe it supports 1.x but requires 2.x branch.

**Reply by mmoss128** [reply]
That is my plan. I was seeing issues with loading Slideshare. Don't have any details yet as I have not fully dug into debugging to see if its an issue on my end.

**Reply by reganlawton** [reply]
The 3.x without looking at the code is basically 98% the same as 2.x but an upgraded version of the core package `embed/embed`. Be sure to check it out if you see any bugs. The version 3.x also supports CraftCMS 4 and 5, probably 6 too I can't remember it's a crazy day.

I'm happy to answer your questions and help with anything else that comes up just fire it here.

**Reply by mmoss128** [reply]
Hi Regan, I got my test environment updated CraftCMS 4 and OEmbed 3.0.9. When using standard URLs for Slideshare all fields parsed comeback as NULL as seen below except for url and code. The code field is not a correct iframe and will cause a connection refused error.  I did notice if I use the URL generated via the share functionality (Pulling the URL out of the iframe it generates) It extracts all of the core information which I need. Now I can update all my entries, but as this was working in the past it seems that something with the embed scrapping has changed and is no longer working. 

Example dd output for entity.media
![image](https://github.com/user-attachments/assets/0c577b13-d4ef-46a7-86ca-6bc87fa0e88d)

**Reply by reganlawton** [reply]
I have more information around this issues. The issues are that Slideshare, while they have oembed support, dont implement it correctly. I have updated the system to handle the meta linking better to use the crawler information of the `embed/embed` package better, as a fallback.

**Reply by reganlawton** [reply]
<img width="1408" alt="Screenshot 2024-09-05 at 5 31 16 PM" src="https://github.com/user-attachments/assets/7ffaa6de-82a0-4105-bc22-ee188183fbee">

Here's an example

I'm currently working on a way to allow for support for [adapters](https://github.com/oscarotero/Embed?tab=readme-ov-file#adapters) and [detectors](https://github.com/oscarotero/Embed?tab=readme-ov-file#detectors) to allow people extend to newer sites

**Reply by mmoss128** [reply]
Nice, that is great to hear! Thank you so much for looking into this.
